cassandra/src/main/scala/geotrellis/spark/io/cassandra/CassandraRDDWriter.scala

```diff
@@ -148,8 +149,13 @@ object CassandraRDDWriter {
                 })
               }
 
-              val results = (rows flatMap elaborateRow flatMap rowToBytes map retire)
-                .join(threads)
+              val results = rows
+                .map(elaborateRow)
+                .parJoin(threads)
```

Comment:
w0000t, i think 

```scala
(rows flatMap elaborateRow flatMap rowToBytes map retire).parJoin(threads)
```

Would be enough! It's already tested and benchmarked long ago, so probably you can just replace all `join`s with `parJoin`s.

Reply:
It makes no sense to do parJoin on any map step, i believe the result would be the same as with a single join.

Reply:
Nvm, lets leave it as you did